From b754049929da5add18a115890bfac1fa9c241c02 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Fri, 6 Nov 2020 10:15:48 +0100
Subject: [PATCH 342/529] dt-bindings: remoteproc: update Hold boot binding for
 stm32 remote processor driver

When the ARM TrustZone is enable, the MCU hold boot is now managed
as a generic reset by the SCMI controller.
Update the bindings according to this new management.
- suppress property related to SMC calls
- add possibility to declare the hold boot as a reset.
- add example with the SCMI server

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: I8c09d2b1dfa283175b6f4e050df3f0fba4edee03
---
 .../bindings/remoteproc/st,stm32-rproc.yaml   | 41 ++++++++++++-------
 1 file changed, 27 insertions(+), 14 deletions(-)

--- a/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
+++ b/Documentation/devicetree/bindings/remoteproc/st,stm32-rproc.yaml
@@ -25,7 +25,14 @@ properties:
     maxItems: 3
 
   resets:
-    maxItems: 1
+    minItems: 1
+    maxItems: 2
+
+  reset-names:
+    items:
+      - const: mcu_rst
+      - const: hold_boot
+    minItems: 1
 
   st,syscfg-holdboot:
     description: remote processor reset hold boot
@@ -36,16 +43,6 @@ properties:
           - description: The offset of the hold boot setting register
           - description: The field mask of the hold boot
 
-  st,syscfg-tz:
-    description:
-      Reference to the system configuration which holds the RCC trust zone mode
-    $ref: "/schemas/types.yaml#/definitions/phandle-array"
-    items:
-      - items:
-          - description: Phandle of syscon block
-          - description: The offset of the trust zone setting register
-          - description: The field mask of the trust zone state
-
   interrupts:
     description: Should contain the WWDG1 watchdog reset interrupt
     maxItems: 1
@@ -135,12 +132,13 @@ required:
   - compatible
   - reg
   - resets
-  - st,syscfg-holdboot
-  - st,syscfg-tz
 
 additionalProperties: false
 
 examples:
+#--------------------
+# Without SCMI server
+#--------------------
   - |
     #include <dt-bindings/reset/stm32mp1-resets.h>
     m4_rproc: m4@10000000 {
@@ -150,9 +148,24 @@ examples:
             <0x38000000 0x10000>;
       resets = <&rcc MCU_R>;
       st,syscfg-holdboot = <&rcc 0x10C 0x1>;
-      st,syscfg-tz = <&rcc 0x000 0x1>;
       st,syscfg-rsc-tbl = <&tamp 0x144 0xFFFFFFFF>;
       st,syscfg-m4-state = <&tamp 0x148 0xFFFFFFFF>;
     };
 
+#--------------------
+# With SCMI server
+#--------------------
+  - |
+    #include <dt-bindings/reset/stm32mp1-resets.h>
+    m4_rproc_w_scmi: m4@10000000 {
+      compatible = "st,stm32mp1-m4";
+      reg = <0x10000000 0x40000>,
+            <0x30000000 0x40000>,
+            <0x38000000 0x10000>;
+      resets = <&scmi_reset RST_SCMI_MCU>,
+               <&scmi_reset RST_SCMI_MCU_HOLD_BOOT>;
+      reset-names = "mcu_rst", "hold_boot";
+      st,syscfg-rsc-tbl = <&tamp 0x144 0xFFFFFFFF>;
+      st,syscfg-m4-state = <&tamp 0x148 0xFFFFFFFF>;
+    };
 ...
